File: lc/mergestringsalternately/mergestrings.java
// here's my solution, it's pretty inelegant
class Solution {
  public String mergeAlternately(String word1, String word2) {
      String ret = "";
      int shorter = word1.length();
      if (word2.length() < word1.length()) {
          shorter = word2.length();
      }
      for(int i = 0; i < shorter; i++) {
          ret += "" + word1.charAt(i) + word2.charAt(i);
      }
      
      if (word1.length() == word2.length()) {
          return ret;
      }
      else if (shorter == word1.length()) {
          ret += word2.substring(shorter, word2.length());
      } else if (shorter == word2.length()) {
          ret += word1.substring(shorter, word1.length());
      }
      return ret;
  }
}

/*
 * here's a really fast (1ms!) solution. 
 * 
 *         StringBuilder result = new StringBuilder();
        int minLength = Math.min(word1.length(),word2.length());
        for(int i=0;i<minLength;i++){
            result.append(word1.charAt(i)).append(word2.charAt(i));
        }
        return result.append(word1.substring(minLength)).append(word2.substring(minLength)).toString();


        StringBuilder in Java represents a mutable sequence of characters 
        (the String Class in Java creates an immutable sequence of characters)
 */

File: lc/easy/mergestringsalternately/mergestrings.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """

        ret = ""
        #one character at a time into the 
        for i, c in enumerate(word1):
            ret = ret + str(c)
            if i < len(word2):
                ret += word2[i]

        if len(word1) < len(word2):
            remainder = word2[len(word1):]
            ret += remainder
        
        return ret
        